### ISSUE TITLE
Allow both `Option=!` and `Option=WrappedType` for optional fields

### ISSUE THREAD

**Issue by RustyNova016** [loganmzz/macon-rs#34]
Currently, you need to choose between whether giving an Option or a value for optional fields. But sometimes, it's easier to give an option to a builder, and others a value.

It would be great to have a `Builder::field_optional()` function when using `Option=WrappedType` that takes in a Option, allowing for both styles depending on the use of the builder

**Comment by loganmzz** [comment]
Sounds good.

I may also add a setting (to be defined) when set makes available following functions:

```rust
// Named
fn field<T: Into<X>>(value: Option<T>) {}
fn field_some<T: Into<X>>(value: T) {}
fn field_none() {}

// Tuple - Unordered
fn set0<T: Into<X>>(value: Option<T>) {}
fn set0_some<T: Into<X>>(value: T) {}
fn set0_none() {}

// Tuple - Ordered
fn set<T: Into<X>>(value: Option<T>) {}
fn some<T: Into<X>>(value: T) {}
fn none() {}
```

As I'm giving a talk (about Rust btw :) ) next week, I won't have time to take a look until next week-end.

**Comment by loganmzz** [comment]
@RustyNova016 Sorry for the delay, I have released [1.3.0](https://github.com/loganmzz/macon-rs/releases/tag/1.3.0), but without setting to reverse optional setters (Option by default and `field_some()` for wrapped type)

**Comment by RustyNova016** [comment]
Neat! Thanks.
